src/main/java/org/jboss/windup/operator/controller/WindupDeploymentController.java

```diff
@@ -0,0 +1,68 @@
+package org.jboss.windup.operator.controller;
+
+import io.fabric8.kubernetes.api.model.OwnerReference;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import lombok.extern.java.Log;
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.windup.operator.model.WindupResource;
+import org.jboss.windup.operator.model.WindupResourceDoneable;
+import org.jboss.windup.operator.model.WindupResourceList;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Log
+@ApplicationScoped
+public class WindupDeploymentController implements Watcher<Deployment> {
+    @Inject
+    MixedOperation<WindupResource, WindupResourceList, WindupResourceDoneable, Resource<WindupResource, WindupResourceDoneable>> crClient;
+
+    @Named("namespace")
+    String namespace;
+
+    @Override
+    public void eventReceived(Action action, Deployment deployment) {
+        log.info("Event " + action + " received for Deployment : " + deployment.getMetadata().getName());
+
+        updateCRStatus(deployment);
+    }
+
+    @Override
+    public void onClose(KubernetesClientException cause) {
+        // TODO Auto-generated method stub
```

Comment:
_Comment to further review the code, NOT mandatory for this PR_
This `onClose` has been left empty.
Worth applying the same logic as in other `WindupController.onClose` method?

Reply:
:+1: